tests_metricflow/sql/compare_data_table.py

```diff
@@ -0,0 +1,159 @@
+from __future__ import annotations
+
+import datetime
+import difflib
+import math
+from dataclasses import dataclass
+from typing import Dict, Optional, SupportsFloat
+
+from metricflow_semantics.mf_logging.pretty_print import mf_pformat_many
+
+from metricflow.data_table.column_types import CellValue
+from metricflow.data_table.mf_table import MetricFlowDataTable
+
+
+def _generate_table_diff_fields(
+    expected_table: MetricFlowDataTable, actual_table: MetricFlowDataTable
+) -> Dict[str, str]:
+    differ = difflib.Differ()
+    expected_table_text = expected_table.text_format()
+    actual_table_text = actual_table.text_format()
+    diff = differ.compare(expected_table_text.splitlines(keepends=True), actual_table_text.splitlines(keepends=True))
+    return {
+        "expected_table": expected_table_text,
+        "actual_table": actual_table_text,
+        "expected_table_to_actual_table_diff": "".join(diff),
+    }
+
+
+@dataclass(frozen=True)
+class DataTableMismatch:
+    """Describes a mismatch in a cell between two tables."""
+
+    message: str
+    row_index: int
+    column_index: int
+    expected_value: CellValue
+    actual_value: CellValue
+
+
+def _check_table_cells_for_mismatch(
+    expected_table: MetricFlowDataTable, actual_table: MetricFlowDataTable
+) -> Optional[DataTableMismatch]:
+    for row_index in range(expected_table.row_count):
+        for column_index in range(expected_table.column_count):
+            # NaNs can't be compared for equality.
+            expected_value = expected_table.get_cell_value(row_index, column_index)
+            actual_value = actual_table.get_cell_value(row_index, column_index)
+            if isinstance(expected_value, SupportsFloat) and isinstance(actual_value, SupportsFloat):
+                if math.isnan(expected_value) and math.isnan(actual_value):
+                    continue
+                if not math.isclose(expected_value, actual_value, rel_tol=1e-6):
+                    return DataTableMismatch(
+                        message="`SupportsFloat` value mismatch",
+                        row_index=row_index,
+                        column_index=column_index,
+                        expected_value=expected_value,
+                        actual_value=actual_value,
+                    )
+            # It should be safe to remove this once `MetricFlowDataTable` is validated as it doesn't allow timezones.
+            elif (
+                isinstance(expected_value, datetime.datetime)
+                and isinstance(actual_value, datetime.datetime)
+                # If expected has no tz but actual is UTC, remove timezone. Some engines add UTC by default.
+                and (actual_value.tzinfo == "UTC" and expected_value.tzinfo is None)
```

Comment:
oof, timezone support is coming, this may get gnarly.....

Those are tomorrow problems, though, for now copying what we've got is good.